src/main.c

```diff
@@ -113,53 +115,53 @@ void NitroMain (void)
     gCoreSys.unk_6C = 1;
     gCoreSys.unk_30 = 0;
 
-    sub_02001028();
+    InitRNG();
     sub_0200AB84();
     sub_02017428();
 
-    Unk_02101D24 = FALSE;
+    gIgnoreCartridgeForWake = FALSE;
 
     while (TRUE) {
-        sub_02000F30();
-        sub_0200106C();
+        CheckHeapCanary();
+        HandleConsoleFold();
         ReadKeypadAndTouchpad();
 
         if ((gCoreSys.heldKeysRaw & RESET_COMBO) == RESET_COMBO) {
-            if (gCoreSys.unk_68 == 0) {
-                sub_02000F60(0);
+            if (gCoreSys.inhibitReset == 0) {
+                SoftReset(0);
             }
         }
 
         if (sub_020349EC()) {
-            sub_02000F30();
+            CheckHeapCanary();
             sub_02000E54();
             sub_0201CDD4(gCoreSys.unk_18);
             sub_0201CDD4(gCoreSys.unk_24);
 
             if (!gCoreSys.unk_30) {
                 OS_WaitIrq(1, OS_IE_V_BLANK);
-                gCoreSys.unk_2C++;
+                gCoreSys.frameCounter++;
             }
```

Comment:
I'm wondering if this is the 30-FPS lock that the gen4 games are infamous for. That `OS_WaitIrq` call looks suspicious.